lib/rpc.ts

```diff
@@ -61,6 +82,29 @@ export async function createReceiverLink(connection: any, receiverOptions: Recei
   };
 }
 
+export async function createReceiverLinkWithHandlers(options: ReceiverLinkOptions): Promise<ReceiverLink> {
+  if (!options.connection) {
+    throw new Error(`Please provide a connection to create the receiver link on a session.`);
+  }
+
+  if (!options.receiverOptions) {
+    throw new Error(`Please provide receiver options.`);
+  }
+  if (!options.onError) {
+    throw new Error(`Please provide onError.`);
+  }
+  if (!options.onMessage) {
+    throw new Error(`Please provide onMessage.`);
+  }
```

Comment:
i'd think those should be closer to the public API (easier to discover when looking at the code to see what went wrong). then use the compiler internally to figure those out.

Reply:
I see. I have them at the public facing API as well. However, I have the habit of validating parameters for internal APIs as well. This helps me catch things during development and testing.